src/tqec/circuit/schedule.py

```diff
@@ -361,19 +361,71 @@ def _compute_number_of_non_virtual_moments(circuit: cirq.Circuit) -> int:
             ]
         )
 
+    def _get_moment_index_by_schedule(self, schedule: int) -> int | None:
+        """Get the index of the moment scheduled at the provided schedule.
+
+        Args:
+            schedule: the schedule at which the Moment instance should be returned.
+                If the schedule is negative, it is considered as an index from
+                the end. For example, -1 is the last schedule and -2 is the
+                last schedule - 1.
```

Comment:
Maybe, to be even more explicit,
```suggestion
                last schedule - 1 (which might not be the second to last schedule).
```